pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -642,7 +643,11 @@ func mergeMetadata(iqrs []*IQR) (*IQR, error) {
 }
 
 func (iqr *IQR) AddColumnsToDelete(cnames map[string]struct{}) {
+	timestampKey := config.GetTimeStampKey()
 	for cname := range cnames {
+		if cname == timestampKey {
+			continue
+		}
```

Comment:
We don't ever want to delete the timestamp field from an IQR? That doesn't seem right. E.g., a `fields -timestamp` command should delete it, right? If we want to enforce this for certain commands like timechart, it would probably be better to enforce it in that command processor instead of in the IQR

Reply:
Ahh, yes got it. I will remove this change.